### PR TITLE
Bump golangci-lint to 1.54.2

### PR DESCRIPTION
**Pull request by zirain**
fixes: #1874

### CODE REVIEW

File: internal/gatewayapi/listener_test.go [hunk 1]
```diff
@@ -45,7 +45,8 @@ func TestProcessTracing(t *testing.T) {
 		},
 	}
 
-	for _, c := range cases {
+	for i := range cases {
```
Comment (arkodg) [inline internal/gatewayapi/listener_test.go:48]: prefer if kept the style same across entire code case and used the existing style of reassigning the var inside the loop i.e.
```
for _, c := range cases {
    c := c
```

Comment (zirain) [reply internal/gatewayapi/listener_test.go]: updated

**Review by arkodg — APPROVED** [review]

**Review by Xunzhuo — COMMENTED** [review]
Can you explain a bit about why these `x := x` lines?

**Comment by zirain** [comment]
see https://stackoverflow.com/questions/62446118/implicit-memory-aliasing-in-for-loop

**Review by AliceProxy — APPROVED** [review]

**Pull request merged**